Bump dependencies to use scritpworker 29.1.0

JohanLorenzo opened the pull request:
Follows https://github.com/mozilla-releng/scriptworker/pull/420 up.

I ran `./maintenance/pin.sh` to get this result.

JohanLorenzo committed:
Bump dependencies to use scritpworker 29.1.0

escapewindow reviewed (approved):
Thanks! We need to track down the test bustage. I’ll try to take a closer look later.

escapewindow committed:
[beetmoverscript] fix tests: avoid circular import

script.py was importing task.py, which was importing script.py. The only
reason task was importing script was to determine `valid_actions`. We
could just drop this check, since `async_main` also ensures we exit on
an invalid action. However, to keep the behavior a bit closer to
existing, I added a `valid_actions` kwarg to `get_task_action`. If
specified, we check the action is a member of `valid_actions`.

escapewindow committed:
[beetmoverscript] fix tests: stop importing scriptworker.test

We have `fake_session` and `fake_session_500` copies in
bouncerscript/tests/conftest.py, so let's copy those over.

escapewindow committed:
[addonscript] fix tests: pin pytest-mock to <1.13

Without this, our mocked aiohttp calls all break. Let's resolve the
underlying issue and unpin in
https://github.com/mozilla-releng/scriptworker-scripts/issues/134 . In
the meantime, pinning should get us green again.

escapewindow committed:
[addonscript] fix tests: see if aioresponses pin helps tc
